nextcloudappstore/core/static/assets/js/img-slider.js

```diff
@@ -0,0 +1,77 @@
+
+var IMG_SLIDER_CONST = {
+    AUTOSCROLL_INTERVAL: 8000 // ms
+};
+
+
+function ImageSlider(element, nextBtnElement, prevBtnElement) {
+    this.element = element;
+    this.strip =
+        new ImageStrip(this, this.element.querySelector('.img-strip'));
+    this.curSlide = 0;
+
+    this.setSlide(this.curSlide);
+
+    var slider = this;
+
+    this.autoScroll = setInterval(function(){slider.increment(1)},
+        IMG_SLIDER_CONST.AUTOSCROLL_INTERVAL);
+
+    nextBtnElement.addEventListener('click', function() {
+        slider.increment(1);
+        clearInterval(slider.autoScroll);
+    });
+    prevBtnElement.addEventListener('click', function() {
+        slider.increment(-1);
+        clearInterval(slider.autoScroll);
+    });
+}
+
+ImageSlider.prototype.setSlide = function(slide) {
+    var imgSpacing = 4;
```

Comment:
It would be awesome if we could go block scoping only. Function scoping creates all sorts of issues, especially with async closures